From c1cadf2e615cbb9af6b7f9edde762b1d3dea9c0c Mon Sep 17 00:00:00 2001
From: "Justin M. Forbes" <jforbes@fedoraproject.org>
Date: Tue, 9 Jan 2024 09:34:44 -0600
Subject: [PATCH] random: replace import_single_range() with import_ubuf()

Upstream commit 6ac805d138709 dropped the iov arg from
import_single_range() and upstream commit 9fd7874c0e5c8 replaced
import_single_range() with import_ubuff().  As our patch 7c0b728ae37e7
still called import_single_range() in drivers/char/random, this needs to
be brought up to date to continue working.

Reported-by: Scott Weaver <scweaver@redhat.com>
Signed-off-by: Justin M. Forbes <jforbes@fedoraproject.org>
---
 drivers/char/random.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/char/random.c b/drivers/char/random.c
index b25f35ff2b65..d860f4f6ba2c 100644
--- a/drivers/char/random.c
+++ b/drivers/char/random.c
@@ -1406,7 +1406,7 @@ SYSCALL_DEFINE3(getrandom, char __user *, ubuf, size_t, len, unsigned int, flags
 	rcu_read_unlock();
 
 	if (rng) {
-		ret = import_single_range(ITER_DEST, ubuf, len, &iov, &iter);
+		ret = import_ubuf(ITER_DEST, ubuf, len, &iter);
 		if (unlikely(ret))
 			return ret;
 		ret = rng->extrng_read_iter(&iter, !!(flags & GRND_RANDOM));
-- 
2.43.0

